cmd/main.go

```diff
@@ -66,10 +70,20 @@ func main() {
 		log.Fatalf(fmt.Sprintf("Failed to load config: %s", err.Error()))
 	}
 
+	nc, err := nats.Connect(cfg.Agent.Server.NatsURL)
+	if err != nil {
+		logger.Warn(fmt.Sprintf("Failed to connect to NATS: %s %s", err, cfg.Agent.Server.NatsURL))
```

Comment:
For me this marks that service is not functional and should be exited. Otherwise it will keep running without heartbeat feature forever, even if NATS came up just a second behind it.